src/download.rs

```diff
@@ -549,3 +572,176 @@ mod error {
         },
     }
 }
+
+#[async_trait]
+trait SnapshotWriteTarget: AsRef<Path> {
```

Comment:
nit: would be nice to add doc comments like this to the trait and the two targets:

```suggestion
/// Shared interface for write targets.
#[async_trait]
trait SnapshotWriteTarget: AsRef<Path> {
```

Reply:
Thanks. I have made the change.